GCTC-NTGC/gc-digital-talent
# Fix PoolCandidate phpunit test

Resolves #3083.

### Notes
Wherever the `assertJson` method is used in phpunit tests to validate an array of objects that use UUIDs, the possibility of test failure occurs because randomly UUIDs cannot be ordered. Instead, create sequential UUIDs for the `id` column using [a new function `integerToUuid`](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3211/commits/2b77ba196d2892fa9aa6dc84385c7e68852ac827) and sort by `id` in the default ascending order.

### Testing Instructions
1. Run `docker-compose exec -w /var/www/html/api php sh -c "./vendor/bin/phpunit --testdox --verbose --repeat 20 tests/Feature/PoolCandidateTest.php"` to run test 20 times.
2. Verify **testFilterByExpiryDate** passes every time.

### Additionally
- [Fixes typo Introspection](https://github.com/GCTC-NTGC/gc-digital-talent/commit/11ab6439bb662100596cc4443a8fd8e541820d57)
- [Moves constant FAR_FUTURE_DATE to laravel config](https://github.com/GCTC-NTGC/gc-digital-talent/commit/8abf72741f593792a165e02af63e6fddcd6b90ac)
- [Updates RefreshDatabase trait](https://github.com/GCTC-NTGC/gc-digital-talent/commit/a46b36d78992a8644c8e5d93de4b44c99cbc4b01)

### Credits
@petertgiles: a PHP paladin!
